VERSIONING.md

```diff
@@ -35,9 +35,14 @@ Backwards incompatible changes to `internal` packages are expected. Versions of
 are expected to be aligned by using BOMs we publish. We will always provide BOMs to allow alignment
 of versions.
 
+Changes may be made that require changes to the an app's dependency declarations aside from just
+incrementing the version on `MINOR` version updates. For example, code may be separated out to a
+new artifact which requires adding the new artifact to dependency declarations.
+
 As a user, if you always depend on the latest version of the BOM for a given `MAJOR` version, and
 you do not use classes in the `internal` package (which you MUST NOT do), you can be assured that
-your app will always function and have the latest features of OpenTelemetry.
+your app will always function and have the latest features of OpenTelemetry without needing any
```

Comment:
It is too much this I think. It means that if you add a new feature like a new sampler people still have to do code changes. I am worried about having the latest features

Reply:
Maybe the latest updates for the features you use

Reply:
I think I meant have access to the new features, reworded a bit

Reply:
The idea being that you're never gated from new features due to dependency hell

Reply:
Yes, I understand; It sounded more like we promise any new feature will be enabled with just gradle work or system properties :)